.github/workflows/Test.yml

```diff
@@ -41,6 +41,15 @@ jobs:
         arch:
           - x64
         allow_failure: [false]
+        is-pr:
+          - ${{ github.event_name == 'pull_request' }}
+        exclude:
+          - is-pr: true
+        include:
+          - version: "1"
+            os: ubuntu-latest
+            arch: x64
+            allow_failure: false
```

Comment:
The implementation seems too convoluted to me.

On a PR, this essentially generates a matrix, then removes everything, and adds back only `{version: 1, os: ubuntu-latest, arch: x64, allow_failure: false}`.  Whereas non-PR events (like a merge, or push) has that entry duplicated. I actually don't know what happens when `include:` has a duplicate, does that count as two separate matrix strategies? Probably not.

I think I would have used [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#creating-a-reusable-workflow) in this situation, the `is-pr:` logic can be simply in `jobs.<job_id>.if:` of the reusable workflow.

*PS:* I also don't like hyphens in yaml key names, strictly speaking they are legal, but in an action keys are sometimes used in expressions, and there it has undefined behaviour.

Reply:
Nice, I didn't know about these. I will look into it and let you know how it goes.

Reply:
I've updated the PR with the Reusable workflow strategy. It should be much more straightforward to understand.
I did not use `if`, I did not see a good way to use it. The only thing I thought of was a verbose conditional that could in theory also be applied to the existing `Test.yml` directly, but I think it would be too complicated, hard to maintain, and ugly.